src/HackerNews/MainPage/MainPage.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useEffect, useState } from "react";
+import NewsSummary from "./NewsSummary/NewsSummary";
+import NewsInfo from "./NewsInfo/NewsInfo";
+
+import { Route, Routes } from "react-router-dom";
+
+import styled from "styled-components";
+
+export interface response {
+  by: string;
+  id: number;
+  descendants: number;
+  kids: Array<number>;
+  score: number;
+  time: number;
+  title: string;
+  type: string;
+  url: string;
+}
+
+const DivWrapper = styled.div`
+  width: 80%;
+  margin: 0 auto;
+  background-color: #e37d62;
+  padding: 30px;
+`;
+
+const RefreshButton = styled.button`
+  width: 40%;
+  height: 35px;
+  margin: 0 auto 10px auto;
+  border-radius: 10px;
+  background-color: #e35d63;
+`;
+
+const ButtonText = styled.p`
+  font-size: 20px;
+`;
+
+const NewsListHeader = styled.div`
+  width: 60%;
+  background-color: #fda06d;
+  border-radius: 10px;
+  padding: 10px;
+  margin: 0 auto 20px auto;
+`;
+
+const HeaderText = styled.p`
+  font-size: 30px;
+  text-align: center;
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const MainPage: () => JSX.Element | null = () => {
+  const url =
+    "https://hacker-news.firebaseio.com/v0/topstories.json?print=pretty";
+  const url_new =
+    "https://hacker-news.firebaseio.com/v0/newstories.json?print=pretty";
+
+  const [news, setNews] = useState<Array<response>>(() => {
```

Comment:
Необязательно тут юзать `getter` для начального значения, проще будет обычной иницилизацией
```suggestion
  const [news, setNews] = useState<Array<response>>([]);
```